zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImpl.java

```diff
@@ -361,4 +393,12 @@ private void validatePassword(String password) throws KeyringException {
             throw new KeyringException(PASSWORD_EMPTY_ERR);
         }
     }
+
+    private void saveKeyring(User user) throws KeyringException {
+        try {
+            users.updateKeyring(user);
+        } catch (IOException ex) {
+            throw new KeyringException(SAVE_USER_KEYRING_ERR, ex);
+        }
```

Comment:
target.keyring() has already been updated before the saveKeyring function is called - is this an expected behaviour?

Reply:
The IOException can occur after the keyring has been updated to the target user- is this an expected behaviour?

Reply:
> target.keyring() has already been updated before the saveKeyring function is called - is this an expected behaviour?

Yes.  `user.updateKeyring()` takes a user object and saves the changes to the keyring field writing them to disk to persist the changes. The method name isn't very clear but it's existing functionality and don't want to mess with it - scope creep. I will rename `users` though to `userService` to make it a bit clearer.